### ISSUE TITLE
Doki-Doki Universe™ [PCSF00379]

### ISSUE THREAD

**Issue by Slade1972** [Vita3K/compatibility#2880]
# App summary
- App name: Doki-Doki Universe™
- App serial: PCSF00379
- App version: 1.00

# Vita3K summary
- Version: v0.2.0
- Build number: 3691
- Commit hash: https://github.com/vita3k/vita3k/commit/31826697
- CPU backend: Dynarmic
- GPU backend: Vulkan

# Test environment summary
- Tested by: Slade 
- OS: Microsoft Windows 10 Professional (build 19045)
- CPU: Intel(R) Core(TM) i7-3770K CPU @ 3.50GHz
- GPU: NVIDIA GeForce GTX 1060 6GB/PCIe/SSE2
- RAM: 16 GB

# Issues
- Text boxes don't render correctly on the Tutorial Planet. This makes selecting the right options impossible. (OpenGL)
- Graphics are good under Vulkan.
- The game utilises the rear touch pad and motion controls for various effects. The rear touch pad is not emulated as far as I'm aware. This prevents the tutorial planet from being completed.

# Screenshots
![ddu1](https://github.com/user-attachments/assets/baa9581d-ad2d-485b-9f59-08540e88f1de)
![ddu2](https://github.com/user-attachments/assets/9918b9cd-00cd-4eba-acb1-63f7b67c5e9a)
![ddu3](https://github.com/user-attachments/assets/fe950532-39f8-4c41-9a8e-e3690bcf31e8)
![ddu4](https://github.com/user-attachments/assets/20b956cb-a770-47bc-b752-ca0f824cbc4c)
![ddu5](https://github.com/user-attachments/assets/57533ddb-426c-40c9-93b0-5b460925258d)

# Log
[vita3k.log](https://github.com/user-attachments/files/18006332/vita3k.log)

# Recommended labels

- Ingame -
- graphics bug
- touchscreen issues

**Comment by benjamintoledo** [comment]
As far as I tested, the rear touchpad is emulated with the right click, and in Android there's a button to change between front and rear.

**Comment by Slade1972** [comment]
> As far as I tested, the rear touchpad is emulated with the right click, and in Android there's a button to change between front and rear.

Yeah, I can see a slightly modified "touch" icon on the screen when I press the right mouse button. I've never seen that before. I don't have a motion capable controller, so I can't fully test the game. It requires both touch and "shake" of the controller.

I'll edit the labels - thanks.

![ddu1](https://github.com/user-attachments/assets/8d5e97a2-f1a3-452d-baee-da2b95678a74)

Do you have a PS4 style controller or one with motion you can use to test the game?

**Comment by benjamintoledo** [comment]
> > As far as I tested, the rear touchpad is emulated with the right click, and in Android there's a button to change between front and rear.
> 
> Yeah, I can see a slightly modified "touch" icon on the screen when I press the right mouse button. I've never seen that before. I don't have a motion capable controller, so I can't fully test the game. It requires both touch and "shake" of the controller.
> 
> I'll edit the labels - thanks.
> 
> ![ddu1](https://private-user-images.githubusercontent.com/11675273/392783775-8d5e97a2-f1a3-452d-baee-da2b95678a74.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM0MTA0MjYsIm5iZiI6MTczMzQxMDEyNiwicGF0aCI6Ii8xMTY3NTI3My8zOTI3ODM3NzUtOGQ1ZTk3YTItZjFhMy00NTJkLWJhZWUtZGEyYjk1Njc4YTc0LmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA1VDE0NDg0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ2YzM1ZDJiNGM0MmJiZDhlNDQ3YmI5ZGVkZTFmNTJiNzNjMWZhMzQzNmVhODA2MTdhMzNhYzg0OGVjZmNjN2UmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.9b31reIyQEsjdq2HQ2FPinyJNoOmLyvy8vhfElXaueE)
> 
> Do you have a PS4 style controller or one with motion you can use to test the game?

I can't afford a PS4 controller for now, so I just can use a generic controller without motion.

**Comment by benjamintoledo** [comment]
> > As far as I tested, the rear touchpad is emulated with the right click, and in Android there's a button to change between front and rear.
> 
> Yeah, I can see a slightly modified "touch" icon on the screen when I press the right mouse button. I've never seen that before. I don't have a motion capable controller, so I can't fully test the game. It requires both touch and "shake" of the controller.
> 
> I'll edit the labels - thanks.
> 
> ![ddu1](https://private-user-images.githubusercontent.com/11675273/392783775-8d5e97a2-f1a3-452d-baee-da2b95678a74.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM2OTQ2MTEsIm5iZiI6MTczMzY5NDMxMSwicGF0aCI6Ii8xMTY3NTI3My8zOTI3ODM3NzUtOGQ1ZTk3YTItZjFhMy00NTJkLWJhZWUtZGEyYjk1Njc4YTc0LmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA4VDIxNDUxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIyNWE2Y2QzMzM1OWZiNzM1MmZjYWJmYmQzNzA5MzA3ZDA1ZTJhOWI5YzVkYTVkYjJkM2M0OWE2ZGYxYzVkMjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.MJq5GQhWCafaFHJamFM5Og-6uLrTsOfm9xd_MS3Dk6Q)
> 
> Do you have a PS4 style controller or one with motion you can use to test the game?

No, but I have tested it on the Android version, and yeah, works.

**Comment by Slade1972** [comment]
The only issue with that is this is for Windows compatibility, not Android. Whilst I appreciate it's the same code base, without proper testing on Windows, I'm not willing to change the labels.  I don't have a PS4 style controller to test with, and given the price of them, I'm not that interested in buying one.

Unfortunately Windows sucks somewhat when it comes to controls with motion capabilities. Windows half expects everything to be an Xbox controller, so 3rd party controllers (that is the cheap Chinese controllers) may have motion control built in, but Windows won't detect it, as the controller says it's Xbox compatible. It's frustrating.

**Comment by benjamintoledo** [comment]
> The only issue with that is this is for Windows compatibility, not Android. Whilst I appreciate it's the same code base, without proper testing on Windows, I'm not willing to change the labels. I don't have a PS4 style controller to test with, and given the price of them, I'm not that interested in buying one.
> 
> Unfortunately Windows sucks somewhat when it comes to controls with motion capabilities. Windows half expects everything to be an Xbox controller, so 3rd party controllers (that is the cheap Chinese controllers) may have motion control built in, but Windows won't detect it, as the controller says it's Xbox compatible. It's frustrating.

Well, now I've got a new cheap controller supposedly with gyro and I'm planning to buy a Dualshock 4, so I'll see later

**Comment by Slade1972** [comment]
> Well, now I've got a new cheap controller supposedly with gyro and I'm planning to buy a Dualshock 4, so I'll see later

I managed to get a Dualsense 5 the other day.  I had some vouchers, and the controller was on sale, so it cost me $20 (AUD). I should go back and check this one. Thanks for mention, it reminded me :)

**Comment by Slade1972** [comment]
# Vita3K summary
- Version: v0.2.0
- Build number: 3703
- Commit hash: https://github.com/vita3k/vita3k/commit/0a3ba020
- CPU backend: Dynarmic
- GPU backend: Vulkan


# Test environment summary
- Tested by: Slade 
- OS: Microsoft Windows 10 Professional (build 19045)
- CPU: Intel(R) Core(TM) i7-3770K CPU @ 3.50GHz
- GPU: NVIDIA GeForce GTX 1060 6GB
- RAM: 16 GB

# Issues
- After an hour or so of game play, the game froze on the summon item screen of a random character. This can be repeated. After a few seconds of additional music playing, the game will crash.
- Motion controls, front / rear touch pad works (multi-touch via Dualsense 5 works).


# Screenshots

![doki8](https://github.com/user-attachments/assets/9211f9d2-ac55-4981-9fee-9d320a8e878f)


# Log
[PCSF00379 - [Doki-Doki Universe™].log](https://github.com/user-attachments/files/18268647/PCSF00379.-.Doki-Doki.Universe.log)


# Recommended labels
- Ingame +
- crash
